File: DynamicKey/AgoraDynamicKey/java/src/main/java/io/agora/sample/EducationTokenBuilder2Sample.java
package io.agora.sample;

import io.agora.education.EducationTokenBuilder2;

public class EducationTokenBuilder2Sample {
    // Need to set environment variable AGORA_APP_ID
    private static String appId = System.getenv("AGORA_APP_ID");
    // Need to set environment variable AGORA_APP_CERTIFICATE
    private static String appCertificate = System.getenv("AGORA_APP_CERTIFICATE");

    private static String roomUuid = "123";
    private static String userUuid = "2882341273";
    private static Short role = 1;
    private static int expire = 600;

    public static void main(String[] args) {
        System.out.printf("App Id: %s\n", appId);
        System.out.printf("App Certificate: %s\n", appCertificate);
        if (appId == null || appId.isEmpty() || appCertificate == null || appCertificate.isEmpty()) {
            System.out.printf("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE\n");
            return;
        }

        EducationTokenBuilder2 tokenBuilder = new EducationTokenBuilder2();
        String roomUserToken = tokenBuilder.buildRoomUserToken(appId, appCertificate, roomUuid, userUuid, role, expire);
        System.out.printf("Education room user token: %s\n", roomUserToken);

        String userToken = tokenBuilder.buildUserToken(appId, appCertificate, userUuid, expire);
        System.out.printf("Education user token: %s\n", userToken);

        String appToken = tokenBuilder.buildAppToken(appId, appCertificate, expire);
        System.out.printf("Education app token: %s\n", appToken);
    }
}

File: DynamicKey/AgoraDynamicKey/cpp/sample/EducationTokenBuilder2Sample.cpp
// Copyright (c) 2014-2024 Agora.io, Inc.

#include <cstdlib>
#include <iostream>

#include "../src/EducationTokenBuilder2.h"

using namespace agora::tools;

int main(int argc, char const *argv[]) {
  (void)argc;
  (void)argv;

  // Need to set environment variable AGORA_APP_ID
  const char *env_app_id = getenv("AGORA_APP_ID");
  std::string app_id = env_app_id ? env_app_id : "";
  // Need to set environment variable AGORA_APP_CERTIFICATE
  const char *env_app_certificate = getenv("AGORA_APP_CERTIFICATE");
  std::string app_certificate = env_app_certificate ? env_app_certificate : "";

  std::string room_uuid = "123";
  std::string user_uuid = "2882341273";
  int16_t role = 1;
  uint32_t expiration_in_seconds = 600;

  std::cout << "App Id:" << app_id << std::endl;
  std::cout << "App Certificate:" << app_certificate << std::endl;
  if (app_id == "" || app_certificate == "") {
    std::cout << "Need to set environment variable AGORA_APP_ID and "
                 "AGORA_APP_CERTIFICATE"
              << std::endl;
    return -1;
  }

  auto room_user_token = EducationTokenBuilder2::BuildRoomUserToken(app_id, app_certificate, room_uuid, user_uuid, role, expiration_in_seconds);
  std::cout << "Education room user token:" << room_user_token << std::endl;

  auto user_token = EducationTokenBuilder2::BuildUserToken(app_id, app_certificate, user_uuid, expiration_in_seconds);
  std::cout << "Education user token:" << user_token << std::endl;

  auto app_token = EducationTokenBuilder2::BuildAppToken(app_id, app_certificate, expiration_in_seconds);
  std::cout << "Education app token:" << app_token << std::endl;

  return 0;
}
